File: 89. Gray Code/89. Gray Code.dart
class Solution {
  List<int> grayCode(int n) {
    int size = 1 << n;  // 2^n
    List<int> result = List<int>.filled(size, 0);
    
    for (int i = 0; i < size; i++) {
      result[i] = i ^ (i >> 1);
    }
    
    return result;
  }
}

File: 89. Gray Code/89. Gray Code.rs
impl Solution {
    pub fn gray_code(n: i32) -> Vec<i32> {
        let size = 1 << n;  // 2^n
        let mut result = vec![0; size as usize];
        
        for i in 0..size {
            result[i as usize] = i ^ (i >> 1);
        }
        
        result
    }
}